### PR TITLE
{tools} [intel-2015a] multitail-6.2.1 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by wpoely86** [comment]
@boegel any reason why we don't enable the UTF-8 version of `ncurses`? ( `--enable-widec` )

File: easybuild/easyconfigs/m/multitail/fix_makefile.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ CONFIG_FILE=$(DESTDIR)/etc/multitail.conf
+ 
+-CC=gcc
++CC:=$(CC)
+ DEBUG=-g -D_FORTIFY_SOURCE=2 # -D_DEBUG # -pg #  -D_DEBUG  #-pg -W -pedantic # -pg #-fprofile-arcs
+ ifeq ($(UTF8_SUPPORT),yes)
+-LDFLAGS+=-lpanelw -lncursesw -lutil -lm $(DEBUG) -rdynamic
+-CFLAGS+=-funsigned-char -D`uname` -O2 -Wall -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\" -DUTF8_SUPPORT
++LDFLAGS:= $(LDFLAGS) -lpanelw -lncursesw -lutil -lm $(DEBUG) -rdynamic
++CFLAGS:= $(CFLAGS) -funsigned-char -D`uname` -O2 -Wall -DVERSION=\"$(VERSION)\" $(DEBUG) -DCONFIG_FILE=\"$(CONFIG_FILE)\" -DUTF8_SUPPORT
```
Comment (boegel) [inline easybuild/easyconfigs/m/multitail/fix_makefile.patch]: if you do it like this, the `-O2` that's there is going to win over whatever `-Ox` `eb` sets, so please include the `$(CFLAGS)` *after* `-O2`

File: easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+sources = [SOURCE_TGZ]
+source_urls = ['http://www.vanheusden.com/multitail/']
+
+patches = ['fix_makefile.patch']
+
+files_to_copy = [
+(['multitail'], 'bin'),
+(['multitail.conf'], 'etc'),
```
Comment (boegel) [inline easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb]: indent is missing here

File: easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+homepage = 'http://www.vanheusden.com/multitail/'
+description = """ MultiTail allows you to monitor logfiles and command output in multiple windows in a terminal, colorize, filter and merge."""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+sources = [SOURCE_TGZ]
+source_urls = ['http://www.vanheusden.com/multitail/']
+
+patches = ['fix_makefile.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb]: better version the patch filename: `multitail-6.2.1_eb.patch`, or atleast mention `multitail` in it

excluding the version might be OK here, it's unlikely that this will break easily for future releases (maybe just use `6`?)

File: easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+files_to_copy = [
+(['multitail'], 'bin'),
+(['multitail.conf'], 'etc'),
+]
+
+prebuildopts = 'DESTDIR=%(installdir)s '
```
Comment (boegel) [inline easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb]: `prebuildopts` after `dependencies`

File: easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+easyblock = 'MakeCp'
+
+name = 'multitail'
+version = '6.2.1'
+
+homepage = 'http://www.vanheusden.com/multitail/'
+description = """ MultiTail allows you to monitor logfiles and command output in multiple windows in a terminal, colorize, filter and merge."""
```
Comment (boegel) [inline easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb]: is this < 120 chars?

**Comment by boegel** [comment]
@wpoely86: I can't think of a reason no to enable UTF-8 in ncurses... not sure if it'll break something 'upstream' (i.e. stuff that depends on it)

**Comment by wpoely86** [comment]
@boegel It's split up in a separate library (`libncursesw`), so it should give any trouble.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/4fb76f90f9422b10a4ec for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/4b4df01b982ab3d34db3 for a full test report.

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb [hunk 6]
```diff
@@ -12,4 +12,7 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = [GNU_SOURCE]
 sources = [SOURCE_TAR_GZ]
 
+# enable UTF-8 version of ncurses
+configopts = '--enable-widec '
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: shouldn't we do this in a separate PR (in *all* ncurses easyconfigs?(

**Comment by boegel** [comment]
@wpoely86: you broke it ;)

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb (hunk 6 above)
Comment (wpoely86) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: first test it here. Then do the rest?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/e1a2be1e331d2e9adbab for a full test report.

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb (hunk 6 above)
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: I'd open a dedicated PR for this, rather than fiddling around in a PR that worked fine without this...

You can start with only touched a single ncurses in the PR, and expand as you go.

Comment (wpoely86) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: no, this PR does not work fine without it. It is needed.

Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: OK, then it makes sense to tackle it here for this ncurses easyconfig, and sync up the others in a dedicated PR.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/07e28c3ccdc593c32998 for a full test report.

**Comment by boegel** [comment]
@wpoely86: closing in...

**Comment by wpoely86** [comment]
@boegel needs more work. On most distro's, the includes are split up: one directory `include/ncurses` for the regular version and one `include/ncursesw` for the utf-8 version. This currently both puts them in `include`. I'm gonna patch the makefile.

**Comment by boegel** [comment]
@wpoely86: it patching the Makefile how distros handle this?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/eb2ad88f1e7cbac90bde for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/f7fa1ca74314f3a01489 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/f3636edb9971324aa0db for a full test report.

**Comment by wpoely86** [comment]
at least! @boegel will give this a spin too?

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb [hunk 7]
```diff
@@ -12,4 +12,11 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = [GNU_SOURCE]
 sources = [SOURCE_TAR_GZ]
 
+configopts = [
+# the regular version
+    '',
+# the UTF-8 enabled version (ncursesw)
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: align comments with the string they apply to please

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb [hunk 8]
```diff
@@ -12,4 +12,11 @@ toolchainopts = {'optarch': True, 'pic': True}
...
+    '--enable-ext-colors --enable-widec --includedir=%(installdir)s/include/ncursesw/'
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: what's this `colors` thing? how does it relate to `--enable-widec`?

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb [hunk 9]
```diff
@@ -12,4 +12,11 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = [GNU_SOURCE]
 sources = [SOURCE_TAR_GZ]
 
+configopts = [
+# the regular version
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: change to `# default build, no custom configure options`

File: easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    (['multitail.conf'], 'etc'),
+]
+
+dependencies = [("ncurses", "5.9")]
+
+prebuildopts = 'DESTDIR=%(installdir)s '
+
+sanity_check_paths = {
+    'files': ['bin/multitail'],
+    'dirs': ["."]
```
Comment (boegel) [inline easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb]: remove this hack, replace with (comma at the end too pls):

```python
    'dirs': [],
```

Comment (wpoely86) [inline easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb]: better update the template then too?

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb (hunk 8 above)
Comment (wpoely86) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-intel-2015a.eb]: it allows some color thingies in UTF-8. Not relevant for a non `widec` build.

**Comment by wpoely86** [comment]
@boegel fixed

File: easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb (hunk 10 above)
Comment (boegel) [inline easybuild/easyconfigs/m/multitail/multitail-6.2.1-intel-2015a.eb]: it's there intentionally in the template, to make the sanity check pass if you're doing `eb --try-software=foo,1.2.3 ...`

**Comment by boegel** [comment]
@wpoely86: I'd like to see a separate PR for syncing the other ncurses easyconfigs with the one in here, before we merge this in... Are you up for that? You can leave the full test report up to me, submitting one for just a couple of easyconfigs you touch is fine by me.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.5, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e0973cdcf629248e8281 for a full test report.

**Comment by wpoely86** [comment]
@boegel ok, will issue PR for ncursesw.

**Comment by boegel** [comment]
PR for others ncurses easyconfigs is in #1337, this one is good to go in, thanks @wpoely86!
